src/components/App/App.jsx

```diff
@@ -12,34 +12,74 @@ import Main from "../Main/Main.jsx";
 import Footer from "../Footer/Footer.jsx";
 import ProtectedRoute from "../ProtectedRoute/ProtectedRoute.jsx";
 import LoginForm from "../LoginForm/LoginForm.jsx";
-import RegisterForm from "../RegisterForm/Registerform.jsx";
+import RegisterForm from "../RegisterForm/RegisterForm.jsx";
 import InfoPopup from "../InfoPopup/InfoPopup.jsx";
 import SavedNews from "../SavedNews/SavedNews.jsx";
 import MobileMenu from "../MobileMenu/MobileMenu.jsx";
 import { CurrentPageContext } from "../../contexts/CurrentPageContext.jsx";
+import { CurrentUserContext } from "../../contexts/CurrentUserContext";
 import { IsLoggedInContext } from "../../contexts/IsLoggedInContext.jsx";
 import { deleteItem, saveItem, getSavedItems } from "../../utils/api.js";
+import * as auth from "../../utils/auth";
 import { getNews } from "../../utils/newsApi.js";
+
 import "./App.css";
 
 function App() {
-  const [isLoggedIn, setIsLoggedIn] = useState(true);
+  const [currentUser, setCurrentUser] = useState(null);
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
   const [activeModal, setActiveModal] = useState("");
   const [currentPage, setCurrentPage] = useState("home");
   const [searchResults, setSearchResults] = useState([]);
   const [isLoading, setIsLoading] = useState(false);
+  const [isLoggedInLoading, setIsLoggedInLoading] = useState(true);
   const [keyword, setKeyword] = useState("");
   const [savedItems, setSavedItems] = useState([]);
   const [searchError, setSearchError] = useState(false);
+  const [serverError, setServerError] = useState({});
+  const navigate = useNavigate();
 
   const handleLogin = ({ email, password }) => {
-    console.log(activeModal);
+    if (!email || !password) {
+      return;
+    }
+    return auth
+      .authorize({ email, password })
+      .then((res) => {
+        setIsLoggedIn(true);
+        localStorage.setItem("jwt", res.token);
+        setCurrentUser({ name: res.name, _id: res._id });
+        closeActiveModal();
+      })
+      .catch((err) => {
+        setServerError({
+          ...serverError,
+          loginError: "Incorrect email or password",
+        });
+        console.error("Failed to login", err);
+      });
   };
 
   const handleRegistration = ({ email, password, name }) => {
```

Comment:
You can create a special variable for watching **loading** states during server requests:

```js
 const [isLoading, setIsLoading] = React.useState(false);

```

And then you can change it to `true` before sending requests and return it back to `false`  in `finally` to stop loading:

And you should pass `isLoading` to every popup to change the **button text**:

```js

 buttonText={isLoading? 'Saving...' : 'Save'}

```